independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/AnnotationStore.java

```diff
@@ -71,163 +50,33 @@ public Collection<AnnotationInstance> getAnnotations(AnnotationTarget target) {
-        Collection<AnnotationInstance> annotations = getOriginalAnnotations(target);
-        List<AnnotationsTransformer> transformers = transformersMap.get(target.kind());
-        if (transformers.isEmpty()) {
-            return annotations;
-        }
-        TransformationContextImpl transformationContext = new TransformationContextImpl(buildContext, target, annotations);
-        for (AnnotationsTransformer transformer : transformers) {
-            transformer.transform(transformationContext);
-        }
-        return transformationContext.getAnnotations();
-    }
-
-    private Collection<AnnotationInstance> getOriginalAnnotations(AnnotationTarget target) {
-        Collection<AnnotationInstance> annotations;
-        switch (target.kind()) {
-            case CLASS:
-                annotations = target.asClass().declaredAnnotations();
-                break;
-            case METHOD:
-                // Note that the returning collection also contains method params annotations
-                annotations = target.asMethod().annotations();
-                break;
-            case FIELD:
-                annotations = target.asField().annotations();
-                break;
-            default:
-                throw new IllegalArgumentException("Unsupported annotation target");
-        }
-
-        return Annotations.onlyRuntimeVisible(annotations);
-    }
-
-    private List<AnnotationsTransformer> initTransformers(Kind kind, Collection<AnnotationsTransformer> transformers) {
-        List<AnnotationsTransformer> found = new ArrayList<>();
-        for (AnnotationsTransformer transformer : transformers) {
-            if (transformer.appliesTo(kind)) {
-                found.add(transformer);
-            }
-        }
-        if (found.isEmpty()) {
-            return Collections.emptyList();
-        }
-        found.sort(BuildExtension::compare);
-        return found;
-    }
-
-    static class TransformationContextImpl extends AnnotationsTransformationContext<Collection<AnnotationInstance>>
-            implements TransformationContext {
-
-        private static final Logger LOG = Logger.getLogger(TransformationContextImpl.class);
-
-        public TransformationContextImpl(BuildContext buildContext, AnnotationTarget target,
-                Collection<AnnotationInstance> annotations) {
-            super(buildContext, target, annotations);
+        Set<DotName> set = new HashSet<>();
+        for (DotName name : names) {
+            set.add(name);
         }
-
-        @Override
-        public Transformation transform() {
-            if (LOG.isTraceEnabled()) {
-                String stack = Arrays.stream(Thread.currentThread().getStackTrace())
-                        .skip(2)
-                        .limit(7)
-                        .map(se -> "\n\t" + se.toString())
-                        .collect(Collectors.joining());
-                LOG.tracef("Transforming annotations of %s %s\n\t...", target, stack);
```

Comment:
Does `AnnotationOverlay` from Jandex provide a similar trace logging? This code together with `AnnotationsTransformationContext.setAnnotations()` was very useful when debugging what transformer changed the final set of annotations...

Reply:
It doesn't, no. Let me take another look at this.

Reply:
Added this logging back (to `Transformation`, which all current annotation transformations use). I'll think about adding similar logging to Jandex, but that's a bit more complex, as it doesn't have a concept of `Transformation` (and `Transformation.done()` in particular :-) ). All transformations are done directly on the `TransformationContext`.